pkg/client/session.go

```diff
@@ -16,6 +16,13 @@ import (
 )
 
 func (c *immuClient) OpenSession(ctx context.Context, user []byte, pass []byte, database string) (err error) {
+	defer func(){
+		if err != nil && err != ErrSessionAlreadyOpen {
+			_ = c.clientConn.Close()
```

Comment:
I'd expect the close the connection after the line where the connection is first established

Reply:
not sure if the error comparison will match `err != ErrSessionAlreadyOpen`

Reply:
1. See my answer above
2. I've already tested: `require.Equal(t, ErrSessionAlreadyOpen, errors.FromError(ErrSessionAlreadyOpen))`